core/src/main/java/com/amplifyframework/core/DeveloperMenuActivity.java

```diff
@@ -16,16 +16,112 @@
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
 import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+
+import com.amplifyframework.util.Time;
 
 /**
  * This is the activity to display the developer menu.
  */
 public final class DeveloperMenuActivity extends Activity {
+    // A device movement is classified as a shake if the acceleration
+    // is above this threshold.
+    private static final double SHAKE_THRESHOLD = 11.7;
+    // The minimum duration (in milliseconds) that the device needs to
+    // be shaken in order to make the developer menu appear or disappear.
+    private static final int SHAKE_TIME = 500;
+
+    // Manager for the device's sensors.
+    private SensorManager sensorManager;
+    // The accelerometer sensor associated with the device.
+    private Sensor accelerometer;
+    // The time (in milliseconds) that the device started shaking
+    // (or 0 if the device is not shaking).
+    private long shakeStart;
+    // The parent layout for the developer menu.
+    private View devMenuLayout;
+
+    // Listen to accelerometer sensor events.
+    private final SensorEventListener sensorEventListener = new SensorEventListener() {
+        @Override
+        public void onSensorChanged(SensorEvent sensorEvent) {
+            float xAccel = sensorEvent.values[0]; // acceleration in the x-direction
+            float yAccel = sensorEvent.values[1]; // acceleration in the y-direction
+            float zAccel = sensorEvent.values[2]; // acceleration in the z-direction
+            double curAcceleration = Math.sqrt(((xAccel * xAccel) + (yAccel * yAccel) + (zAccel * zAccel)));
+            if (curAcceleration > SHAKE_THRESHOLD) {
+                long currentTime = Time.now();
+                if (shakeStart == 0) {
+                    shakeStart = currentTime;
+                } else if (currentTime - shakeStart > SHAKE_TIME) {
+                    changeVisibility();
+                    shakeStart = 0;
+                }
+            }
+        }
+
+        @Override
+        public void onAccuracyChanged(Sensor sensor, int accuracy) { }
+    };
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_dev_menu);
+        sensorManager = (SensorManager) getSystemService(Context.SENSOR_SERVICE);
+        if (sensorManager != null) {
+            accelerometer = sensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER);
+        }
+        devMenuLayout = findViewById(R.id.devLayout);
+        devMenuLayout.setFocusable(true);
+        devMenuLayout.setVisibility(View.GONE);
+        shakeStart = 0;
+    }
+
+    @Override
+    protected void onResume() {
+        if (accelerometer != null) {
+            sensorManager.registerListener(sensorEventListener, accelerometer, SensorManager.SENSOR_DELAY_NORMAL);
```

Comment:
If I'm reading this correctly, Amplify would keep the accelerometer active at all times, to receive sensor data, so it can detect a shake. This will drain the battery, so we need a different solution. What mechanisms can you put in place to limit the lifecycle of the accelerometer? How can you prevent this from running "most of the time", when Amplify is in production?

Reply:
When you say "Amplify is in production", are you referring to when the developer is using Amplify in their app and their app is in production (release build)?

Reply:
The second one!

Reply:
The developer menu will only be activated for a debuggable build, not for release builds (I added a check in Amplify.java in a previous PR that only starts the developer menu activity if the build is debuggable).

Reply:
Okay, great. Thanks for reminding me. It's my job to be paranoid about it. :-D